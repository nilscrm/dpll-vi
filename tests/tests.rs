use insta_cmd::assert_cmd_snapshot;
use rstest::rstest;
use std::{fs::File, path::{Path, PathBuf}, process::Command};

fn test_cnf_files(group_name: &str, path: &Path) {
  insta::with_settings!({
    snapshot_path => format!("snapshots/{}", group_name),
    filters => [(r"Performance\s+Time\s+.*\sms\s+Speed.*IPS", "[Performance...]")] },
  {
    let stdin = File::open(path).unwrap();
    assert_cmd_snapshot!(path.file_stem().unwrap().to_str().unwrap(), Command::new("vine").args(["run", "dpll/main.vi"]).arg("--breadth-first").stdin(stdin));
  });
}

#[rstest]
fn test_minimal(#[files("tests/minimal/*.cnf")] path: PathBuf) {
  test_cnf_files("minimal", &path);
}

#[rstest]
fn test_medium(#[files("tests/medium/*.cnf")] path: PathBuf) {
  test_cnf_files("medium", &path);
}

#[rstest]
fn test_satlib_sat(#[files("tests/uf5-218_first_10/*.cnf")] path: PathBuf) {
  test_cnf_files("uf5-218_first_10", &path);
}

#[rstest]
fn test_satlib_unsat(#[files("tests/uuf50-218_first_10/*.cnf")] path: PathBuf) {
  test_cnf_files("uuf50-218_first_10", &path);
}
