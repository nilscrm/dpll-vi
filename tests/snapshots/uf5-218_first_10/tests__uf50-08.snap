---
source: tests/tests.rs
info:
  program: vine
  args:
    - run
    - dpll/main.vi
    - "--breadth-first"
---
success: true
exit_code: 0
----- stdout -----
satisfiable
evaluation under model: true
var 1 = true
var 2 = false
var 3 = false
var 4 = false
var 5 = false
var 6 = true
var 7 = false
var 8 = true
var 9 = true
var 10 = true
var 11 = true
var 12 = true
var 13 = true
var 14 = false
var 15 = false
var 16 = true
var 17 = false
var 19 = true
var 20 = false
var 21 = false
var 22 = true
var 23 = false
var 24 = false
var 25 = true
var 26 = true
var 27 = false
var 28 = true
var 29 = false
var 30 = true
var 31 = true
var 32 = true
var 33 = true
var 34 = false
var 35 = false
var 36 = false
var 37 = false
var 38 = false
var 39 = false
var 40 = true
var 41 = true
var 42 = true
var 43 = false
var 44 = false
var 45 = false
var 46 = false
var 47 = false
var 48 = true
var 49 = true
var 50 = true

----- stderr -----

Interactions
  Total           42_215_937
    Depth         21_622_754
    Breadth                1
  Annihilate      23_830_920
  Commute             67_247
  Copy             4_042_675
  Erase            4_824_124
  Expand           3_350_691
  Call             3_862_102
  Branch           2_238_178

Memory
  Heap               670_816 B
  Allocated      935_731_968 B
  Freed          935_731_968 B

[Performance...]
